README.md

```diff
@@ -0,0 +1,701 @@
+│                  │   │                  │
+│      PHOTO       │   │      PHOTO       │
+│                  │   │                  │
+│                  │   │                  │
+└──────────────────┘   └──────────────────┘
+
+owner=claire───────┐   owner=diana────────┐
+me:read            │   me:read            │
+│                  │   │                  │
+│      PHOTO       │   │      PHOTO       │
+│                  │   │                  │
+│                  │   │                  │
+└──────────────────┘   └──────────────────┘
+```
+
+# System context assumptions
+
+We assume a system where resources can be referenced by unique identifiers (for example, via content hashes) and messages between peers are signed using public key cryptography. Optionally, messages can form append-only logs (by including sequence number and backlinks) so as to provide verifiable causal ordering and stricter capability condition restrictions.
+
+# Definitions
+
+**peer:** a user possessing knowledge of the private key for an ed25519 signing keypair; publishes operations and syncs document state with other peers  
+**owner:** the peer who created a document  
+**issuer:** the peer issuing authority via a capability  
+**receiver:** the peer receiving authority via a capability  
+**authorizer:** the peer authorizing a request or operation against a capability  
+**invoker** the peer attempting to use the authority a capability delegates (by publishing operations or syncing documents)  
+**document:** a uniquely identifiable resource which capabilities are associated with  
+**operation:** messages peers publish to the network containing changes to documents, require that suitable authority is demonstrated before being accepted by another peer  
+**collection:** a collection of documents  
+**group:** a group of authors (public keys) which can be the owner of a document, and receiver of a capability  
+**schema:** description of a documents' format, is addressable by a unique schema id  
+
+# Capability
+
+| Field        | Type                          | Required | Description                                             |
+| ------------ | ----------------------------- | -------- | ------------------------------------------------------- |
+| `issuer`     | `PublicKey`                   | Yes      | Public key of the issuing author                        |
+| `receiver`   | `PublicKey` / `GroupId` / `*` | Yes      | The receiver of this capability                         |
+| `subject`    | `PublicKey`                   | Yes      | The initial (root) issuer in any capability chain       |
+| `action`     | `String`                      | Yes      | The action a capability gives authority to perform      |
+| `conditions` | `Conditions`                  | Yes      | Conditions which restrict this capability               |
+| `not_before` | `Integer`                     | No       | "Not before" UTC Unix Timestamp in seconds (valid from) |
```

Comment:
:fire: danger, I think this is a soft requirement and is just inviting attackers to lie?
you can catch them lying if you spot them "time travelling" (timestamps not monotonically increasing in causal DAG)

Reply:
Same with expires I guess

Reply:
`not_before` and `expires` are not related to operation timestamps (this is in `conditions) but the valid lifetime of a capability token. No peer (authorizer) should accept a capability which has expires, even if the operation being authorized would have been accepted under the contained conditions.

The interplay between these two sets of timestamp restrictions is very interesting!

Reply:
`not_before` and `expires` are measured against system clock when requests occur, `from_timestamp` and `to_timestamp` relate to a operation timestamps.

Reply:
And if you want operation restrictions to not rely on timestamps at all, the you can use `from_seq` and `to_seq` which relates to positions in an authors document contribution log.